dependabot.yml: dev-dependencies group sweeps in majors (breaks npm ci), and every ecosystem label is missing

Two defects in `.github/dependabot.yml`, both visible within an hour of the npm ecosystem going live in #1519. One is mine, one predates it.

### 1. The `dev-dependencies` group sweeps in major bumps

#1525 opened as **19 dev-dependencies in one PR**, including:

| package | from | to | |
|---|---|---|---|
| `typescript` | ^5.7.2 | **^7.0.2** | two majors |
| `@sveltejs/vite-plugin-svelte` | ^5.0.1 | **^7.2.0** | two majors |
| `@types/node` | ^25.6.0 | **^26.1.1** | major |

It fails before a single test runs:

```
npm error code ERESOLVE
While resolving: @sveltejs/kit@2.69.3
Found: typescript@7.0.2
Could not resolve dependency:
  peerOptional typescript@"^5.3.3 || ^6.0.0" from @sveltejs/kit@2.69.3
```

TypeScript 7 is outside SvelteKit 2.69.3's peer range, so `npm ci` can't resolve the tree at all.

The config bounds the production group to minor/patch but leaves dev unbounded:

```yaml
groups:
  dev-dependencies:
    dependency-type: development        # <- no update-types: majors ride in
  production-minor-patch:
    dependency-type: production
    update-types: [minor, patch]        # <- bounded
```

That asymmetry was an oversight, not a decision — the intent was that majors surface individually and justify themselves, which is exactly why #1381 and #1383 were kept out of batch merges. Adding `update-types: [minor, patch]` to the dev group restores it, and majors get their own PRs.

Worth deciding rather than assuming: whether one `dev-dependencies` group is the right shape at all. 19 packages in a PR is reviewable only while they're all patch bumps; splitting by concern (build toolchain / lint / test) may age better.

### 2. Every ecosystem label is missing

Dependabot comments this on **every PR it opens**, and has been for months:

```
### Labels
The following labels could not be found: `maven`. Please create it before Dependabot can add it
The following labels could not be found: `github-actions`.
```

The config asks for `[dependencies, maven]`, `[dependencies, github-actions]` and `[dependencies, npm]`. The repo has **`dependencies`** and **`java`** — but no `maven`, `github-actions`, or `npm`. So the ecosystem labelling silently never happens and nothing is filterable by ecosystem.

I inherited this in #1519 by copying the existing pattern without checking the labels existed.

Two ways to fix, and it's a real choice: create the three labels, or point the config at what's already there. The existing **`java`** label suggests the original intent may have been `java` rather than `maven` — worth confirming before creating a near-duplicate.

--- a/.github/dependabot.yml
+++ b/.github/dependabot.yml
@@ -13,17 +13,65 @@ updates:
     open-pull-requests-limit: 3
     labels: [dependencies, github-actions]
   # saiku-ui has its own package.json and is not part of the Maven reactor, so
-  # it needs its own entry — without one it is never scanned at all. Grouped so
-  # a week's worth of SvelteKit churn arrives as a couple of PRs, not a wall.
+  # it needs its own entry — without one it is never scanned at all.
+  #
+  # Every group below is bounded to minor/patch deliberately. Majors have to
+  # surface as their own PR and justify themselves — that is why docker/login-action
+  # v4 (#1381) and hamcrest 3.0 (#1383) are still sitting on their own rather than
+  # riding in on a batch. Leaving the dev group unbounded broke exactly that rule:
+  # it swept typescript 5->7 and @sveltejs/vite-plugin-svelte 5->7 into a
+  # 19-package PR (#1525) that npm ci could not even resolve, because typescript 7
+  # is outside @sveltejs/kit's peerOptional range. Bounding does NOT hold back
+  # security fixes — these groups are version updates only (`applies-to` defaults
+  # to version-updates) and Dependabot's security updates run on a separate path.
+  #
+  # Split by blast radius rather than one flat "dev" bucket, because a red PR in
+  # each means a different thing:
+  #   - build-toolchain gates `npm run build`, which saiku-webapp's
+  #     frontend-maven-plugin runs during `mvn verify` — a break here takes the
+  #     whole reactor down on development, not just the UI.
+  #   - dev-tooling is what CI's `ui` job actually exercises (svelte-check,
+  #     vitest, eslint).
+  #   - storybook is not built by CI at all and its packages move in lockstep, so
+  #     it can churn on its own without masking the other two.
+  # Order matters: a dependency lands in the FIRST group it matches, so the
+  # pattern groups must come before the catch-all.
   - package-ecosystem: npm
     directory: /saiku-ui
     schedule:
       interval: weekly
-    open-pull-requests-limit: 5
+    # Raised from 5. Now that the dev group is bounded, each pending major
+    # (typescript, @sveltejs/vite-plugin-svelte, @types/node) gets its own PR and
+    # competes with the grouped ones for this pool; at 5 the majors would starve
+    # and we would be back to never seeing them.
+    open-pull-requests-limit: 10
     labels: [dependencies, npm]
     groups:
-      dev-dependencies:
+      build-toolchain:
         dependency-type: development
+        update-types: [minor, patch]
+        patterns:
+          - svelte
+          - svelte-check
+          - "@sveltejs/*"
+          - vite
+          - tailwindcss
+          - "@tailwindcss/*"
+          - typescript
+          - "@types/node"
+      storybook:
+        dependency-type: development
+        update-types: [minor, patch]
+        patterns:
+          - storybook
+          - "@storybook/*"
+          - "@chromatic-com/*"
+      # Catch-all for the rest of devDependencies (eslint, typescript-eslint,
+      # globals, vitest, playwright, jsdom/happy-dom) and for anything added
+      # later, so a new dev dep is never orphaned into a PR of its own.
+      dev-tooling:
+        dependency-type: development
+        update-types: [minor, patch]
       production-minor-patch:
         dependency-type: production
         update-types: [minor, patch]

